### PR TITLE
Feat/ease job description

### PR DESCRIPTION
**Pull request by cvillamer13**
- Create job
- retrieving a job Company
- Test retrieving all jobs
- updating a job
-  deleting a job
- create ORM Retrieve Data with where 
- create API for job Suggestion

### CODE REVIEW

File: src/services/job.service.ts [hunk 1]
```diff
@@ -172,15 +167,23 @@ export async function retrieveAllJob() {
...
-//       where('verified', '==', true)
-//     )
-//   ).then((results) => {
-//     let x: any[] = []
-//     results.forEach((i) => x.push([i.id, i.data()]))
-//     return x
-//   })
-// }
+
+export async function jobSuggestionServ(data:any){
```
Comment (aldenvallestero) [inline src/services/job.service.ts:171]: jobSuggestion

File: src/services/job.service.ts [hunk 2]
```diff
@@ -83,31 +74,36 @@ export async function createJob(data: ICreateJob, uid:any) {
...
 
-export async function retrieveJob(data: string) {
-  return await getDoc(doc(getFirestore(firebase), 'jobs', data))
-    .then((result) => {
+export async function retrieveJob(data: IRetrieveJobCompany) {
+  return await firestore
+    .retrieveWhere(data)
+    .then((Result) => {
+      const x: any[] = []
+      Result.forEach((i: any) => x.push([i.id, i.data()]))
```
Comment (aldenvallestero) [inline src/services/job.service.ts:82]: x.push({ id: i.id, ...data() })

File: src/controllers/job.controller.ts [hunk 3]
```diff
@@ -1,22 +1,23 @@
 import { Request, Response } from 'express'
-import { tokenValidation } from '../middlewares/token.validation.middleware'
+import { tokenValidation } from '../middlewares/tokenValidation.middleware'
 import {
   createJob as CreateJob,
   retrieveJob as RetrieveJob,
   updateJob as UpdateJob,
   deleteJob as DeleteJob,
   retrieveAllJob as RetrieveAllJob,
+  jobSuggestionServ
```
Comment (aldenvallestero) [inline src/controllers/job.controller.ts:9]: jobSuggestion

File: src/configs/interfaces.config.ts [hunk 4]
```diff
@@ -126,20 +129,24 @@ export interface UIDRequest extends Response {
...
-  Qualififications : any,
-  Bonuses:any,
-  Experince:any,
-  
-
-
+  salary: string | number
+  position: string
+  responsibilities: any
+  skills: any
```
Comment (aldenvallestero) [inline src/configs/interfaces.config.ts:136]: string[]

File: src/configs/interfaces.config.ts [hunk 5]
```diff
@@ -126,20 +129,24 @@ export interface UIDRequest extends Response {
...
+  jobTypes: any
+  Schedules: any
```
Comment (aldenvallestero) [inline src/configs/interfaces.config.ts:138]: string

File: src/configs/interfaces.config.ts [hunk 6]
```diff
@@ -126,20 +129,24 @@ export interface UIDRequest extends Response {
...
+  JobDescription: string
+  Qualififications: any
+  Bonuses: any
+  Experince: any
```
Comment (aldenvallestero) [inline src/configs/interfaces.config.ts:142]: experiences: string[]

**Review by cvillamer13 — COMMENTED** [review]
Done

File: src/configs/firebase.config.ts [hunk 7]
```diff
@@ -93,13 +94,20 @@ const firestore = {
...
       })
   },
+  retrieveWhere: async (data: IRetrieveWhere) => {
+    return await firebaseAdmin
+      .firestore()
+      .collection(data.collection)
+      .where(data.where.column, data.where.expression, data.where.value)
+      .get()
+      .then((result: any) => {
+        return result
```
Comment (aldenvallestero) [inline src/configs/firebase.config.ts:104]:
```suggestion
        return result
```

File: src/configs/interfaces.config.ts [hunk 8]
```diff
@@ -113,7 +113,10 @@ export interface IDelete {
   collection: string
   id: string
 }
-
+export interface IRetrieveWhere {
+  collection: any
```
Comment (aldenvallestero) [inline src/configs/interfaces.config.ts:117]:
```suggestion
  collection: string
```

File: src/configs/interfaces.config.ts (hunk 4 above)
Comment (aldenvallestero) [inline src/configs/interfaces.config.ts:136]:
```suggestion
  skills: string[]
```

**Review by aldenvallestero — APPROVED** [review]

**Pull request merged**